### DISCUSSION TITLE
Promblem with BulkMessenger

### DISCUSSION THREAD

**Discussion by rspov** [nutgram/nutgram#311] [Q&A]
Could you please suggest what I am doing wrong?

Warning: Undefined array key "TOKEN" in /var/www/vhosts/west.space/apps.west.space/bulk_messenger.php on line 7

Fatal error: Uncaught TypeError: SergiX44\Nutgram\Nutgram::__construct(): Argument #1 ($token) must be of type string, null given, called in /var/www/vhosts/west.space/apps.westking.space/bulk_messenger.php on line 7 and defined in /var/www/vhosts/westspace/apps.west.space/vendor/nutgram/nutgram/src/Nutgram.php:90 Stack trace: #0 /var/www/vhosts/west.space/apps.west.space/bulk_messenger.php(7): SergiX44\Nutgram\Nutgram->__construct() #1 {main} thrown in /var/www/vhosts/west.space/apps.west.space/vendor/nutgram/nutgram/src/Nutgram.php on line 90

**Comment by Lukasss93** [comment]
Just set your bot token inside the Nutgram instance:

```php
$bot = new Nutgram('your-token');
```

For more info, see the doc: https://nutgram.dev/docs/usage/bulk_messenger

**Reply by rspov** [reply]
> Just set your bot token inside the Nutgram instance:
> 
> ```
> $bot = new Nutgram('your-token');
> ```
> 
> For more info, see the doc: https://nutgram.dev/docs/usage/bulk_messenger

Did not work(

Warning: Undefined array key "561111127:TEST3SbCvTEST-apmS3TPGdG_HccKFWgZ_A" in /var/www/vhosts/west.space/apps.west.space/bulk_messenger.php on line 7

Fatal error: Uncaught TypeError: SergiX44\Nutgram\Nutgram::__construct(): Argument https://github.com/nutgram/nutgram/pull/1 ($token) must be of type string, null given, called in /var/www/vhosts/west.space/apps.west.space/bulk_messenger.php on line 7 and defined in /var/www/vhosts/west.space/apps.west.space/vendor/nutgram/nutgram/src/Nutgram.php:90 Stack trace: #0 /var/www/vhosts/west.space/apps.west.space/bulk_messenger.php(7): SergiX44\Nutgram\Nutgram->__construct() https://github.com/nutgram/nutgram/pull/1 {main} thrown in /var/www/vhosts/west.space/apps.west.space/vendor/nutgram/nutgram/src/Nutgram.php on line 90


bulk_messenger.php
```
<?php

use SergiX44\Nutgram\Nutgram;

require 'vendor/autoload.php';

$bot = new Nutgram($_ENV['561111127:TEST3SbCvTEST-apmS3TPGdG_HccKFWgZ_A']);
```

**Reply by Lukasss93** [reply]
`$_ENV['TOKEN']`  is just an example...

You can insert the token in this way:
```php
$bot = new Nutgram('561111127:TEST3SbCvTEST-apmS3TPGdG_HccKFWgZ_A');
```
